lib/state_machine/dsl.ex

```diff
@@ -269,9 +274,55 @@ defmodule StateMachine.DSL do
       end
 
       def trigger_with_context(model, event, payload \\ nil) do
-        Context.build(__state_machine__(), model)
-        |> Event.trigger(event, payload)
+        context = Context.build(__state_machine__(), model)
+
+        Event.trigger(context, event, payload)
+        |> case do
+          %{status: :failed} = result -> result
+
+          result ->
+            AfterAllEvents.callbacks(context, payload)
+            result
```

Comment:
Looks fine. I think we do not need case condition here because you already have 
```
def callbacks(%{status: :failed} = ctx, _arg), do: ctx
```
pattern inside `StateMachine.AfterAllEvents` module.